public/app/plugins/datasource/prometheus/metric_find_query.ts

```diff
@@ -48,7 +48,8 @@ export default class PrometheusMetricFindQuery {
 
     const labelValuesQuery = this.query.match(labelValuesRegex);
     if (labelValuesQuery) {
-      if (labelValuesQuery[1]) {
+      const hasEmptyFilter = () => labelValuesQuery[1].split(' ').join('') === '{}';
```

Comment:
I think this should be outside of this method, ideally a helper function outside of the class.

Reply:
I moved it out, and made a few things more descriptive.

Reply:
I also ensured that the tests perform the various tests cases in the array it iterates (vs. just using that to make test descriptions)
- https://github.com/grafana/grafana/pull/78242/commits/4861ad7a29de94a0a289a734ce92fa14b78ff11a